scripts/packaging/foundationpreupm.ps1

```diff
@@ -0,0 +1,39 @@
+<#
+.SYNOPSIS
+    Prepares the MRTK foundation for UPM packaging.
+.DESCRIPTION
+    Prepares the MRTK foundation for UPM packaging.
+.PARAMETER PackageRoot
+    The root folder containing the foundation package contents. If not specified, the current folder is presumed.
+#>
+param(
+    [string]$PackageRoot
+)
+
+if (-not $PackageRoot) {
+    throw "Missing required parameter: -PackageRoot."
+}
+
+# Ensure the required folder exists
+$samplesFolder = "$PackageRoot\Samples~"
+if (-not (Test-Path -Path $samplesFolder)) {
+    New-Item $samplesFolder -ItemType Directory | Out-Null
+}
+
+# Copy the scene system's default scenes to Samples~
+Copy-Item -Path "$PackageRoot/SceneSystemResources" -Destination "$samplesFolder/SceneSystemResources" -Recurse -Force
+
+# Create the samples data for the package.json file
+$samples = "`"samples`": ["
+$samples = $samples + "`n      {`n"
+$samples = $samples + "        `"displayName`": `"Scene System Resources`",`n"
+$samples = $samples + "        `"description`": `"Resources used by the MRTK Scene System`",`n"
+$samples = $samples + "        `"path`": `"Samples~/SceneSystemResources`"`n"
+$samples = $samples + "      }"
+$samples = $samples + "`n   ]"
+
+# Update the project.json file to specify the samples contanined in the package
+$packageJsonPath = "$PackageRoot/package.json"

```

Comment:
This file is kinda a copy paste of the extensionsfolderpreupm one, where's there's a ton of overlap - the delta is the displayname/description/path/folderstocopy, but these all seem like things would be config where the rest of the file should be execution.

I think the next time we'd want to add another such thing, we should do the work to refactor this so that we don't copy pasta more.

Reply:
agreed. i was thinking that as i diverged this one a bit (it is more hard coded than the others)